src/internal/connector/graph/service.go

```diff
@@ -22,16 +22,21 @@ import (
 )
 
 const (
-	logGraphRequestsEnvKey  = "LOG_GRAPH_REQUESTS"
-	numberOfRetries         = 3
-	retryAttemptHeader      = "Retry-Attempt"
-	retryAfterHeader        = "Retry-After"
-	defaultMaxRetries       = 3
-	defaultDelay            = 3 * time.Second
-	absoluteMaxDelaySeconds = 180
-	rateLimitHeader         = "RateLimit-Limit"
-	rateRemainingHeader     = "RateLimit-Remaining"
-	rateResetHeader         = "RateLimit-Reset"
+	logGraphRequestsEnvKey = "LOG_GRAPH_REQUESTS"
+	retryAttemptHeader     = "Retry-Attempt"
+	retryAfterHeader       = "Retry-After"
+	defaultMaxRetries      = 3
+	defaultDelay           = 3 * time.Second
+	httpTimeout            = time.Minute * 3
+
+	// absoluteMaxDelay is used to define the maximum delay for all
+	// the retries. Extra 1 minute to account for any other delays
+	// outside of http.Timeout
+	absoluteMaxDelay = defaultMaxRetries*httpTimeout + 1*time.Minute
```

Comment:
what happens if someone sets max-retires to, say, 10?  Is this better off as a func instead of a const?

Reply:
I think we should still ideally compute it, unless we want to ceil to a max time duration.